core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -225,6 +225,9 @@ private TableProperties() {
   public static final String MIN_SNAPSHOTS_TO_KEEP = "history.expire.min-snapshots-to-keep";
   public static final int MIN_SNAPSHOTS_TO_KEEP_DEFAULT = 1;
 
+  public static final String DYNAMIC_OVERWRITE_ISOLATION_LEVEL = "write.dynamic.overwrite.isolation-level";
```

Comment:
Are we sure we want to have a table property for this? Also, does this mean it will be a breaking change in the behavior given the currently proposed default?

I am not sure what's best yet but we can express the same things via data source options.

Reply:
Yea, Flink does use this API.

One option is having sparkWriteConf, something like dynamic.overwrite.{table}.isolation.level?  Which then activate if paired with "validate-from-snapshot".  Not sure if that's the best way.  Didn't change yet so we can discuss

Reply:
What I mean is that we can use Dataset options for this. We anyway need to pass the snapshot ID that was scanned to make it work and the only way to do that is through options. Even if we have a property, we still need to set the option. I'd probably use options to begin with. I can be convinced otherwise, though.